ros/api/v1/hosts.py

```diff
@@ -376,57 +380,138 @@ def get(self, host_id):
 
 class ExecutiveReportAPI(Resource):
 
+    systems_per_state = {
+        "under_pressure": fields.Raw,
+        "undersized": fields.Raw,
+        "oversized": fields.Raw,
+        "waiting_for_data": fields.Raw,
+        "idling": fields.Raw
+    }
+    condtions = {
+        "cpu": fields.Raw,
+        "io": fields.Raw,
+        "memory": fields.Raw
+    }
+    meta = {
+        "total_count": fields.Integer,
+        "non_optimized_count": fields.Integer,
+        "conditions_count": fields.Integer,
+    }
+    report_fields = {
+        "systems_per_state": fields.Nested(systems_per_state),
+        "conditions": fields.Nested(condtions),
+        "meta": fields.Nested(meta)
+    }
+
+    @marshal_with(report_fields)
     def get(self):
+        account_number = identity(request)['identity']['account_number']
+        system_queryset = system_ids_by_account(account_number, fetch_records=True)
+
+        # System counts
+        all_systems_count = system_queryset.count()
+        under_pressure_systems = count_per_state(system_queryset, {'state': "Under pressure"})
+        undersized_systems = count_per_state(system_queryset, {'state': "Undersized"})
+        oversized_systems = count_per_state(system_queryset, {'state': "Oversized"})
+        awaiting_data_systems = count_per_state(system_queryset, {'state': "Waiting for data"})
+        idling_systems = count_per_state(system_queryset, {'state': "Idling"})
+
+        non_optimized = system_queryset.filter(
+            or_(
+                System.cpu_state is not None,
+                System.io_state is not None,
+                System.memory_state is not None,
+            ),
+            and_(
+                System.state != 'Optimized',
+                System.state != 'Waiting for data'
+            )
```

Comment:
@kgaikwad Does this make sense to you? :thinking:

Reply:
@saltgen  - instead of these multiple conditions, does it make more sense to check if system state is any of `"Oversized", "Undersized", "Idling", "Under pressure"` then it is non optimized one?

Reply:
@kgaikwad Yes that could work. 

My concern is that state and substate data are both coming from an external source, so I think just to be completely sure we should also check the sustates in addition to states.


The resulting query is something like this, it is longer.

```
non_optimized = system_queryset.filter(
            or_(
                System.cpu_state is not None,
                System.io_state is not None,
                System.memory_state is not None,
            ),
            and_(
                System.state == 'Oversized',
                System.state == 'Undersized',
                System.state == 'Idling',
                System.state == 'Under pressure',
            )
        )

```
Thoughts?

Reply:
@saltgen  - could you share any scenario where system will hit substates & not main state?

Reply:
Even if states are coming from advisor engine, a system ideally has to hit a main state always. Having said that, IMHO we can think of checking the main state and then consider checking any sub-state.

Reply:
@kgaikwad Based on discussions this is the current query, change pushed.

```
non_optimized = system_queryset.filter(
            and_(
                System.state != 'Optimized',
                System.state != 'Waiting for data'
            )
        )
```

Reply:
@kgaikwad Based on discussions this is the current query, change pushed.

```
non_optimized = system_queryset.filter(
            and_(
                System.state != 'Optimized',
                System.state != 'Waiting for data'
            )
        )
```

Reply:
@kgaikwad Latest query,

```
non_optimized = system_queryset.filter(
            or_(
                System.state == 'Oversized',
                System.state == 'Undersized',
                System.state == 'Idling',
                System.state == 'Under pressure'
            )
        )
```